### PR TITLE
Expanding the intro to step 51

### PR DESCRIPTION
**Pull request by aberleider**
Added some references to the intro of step 51

### CODE REVIEW

**Comment by drwells** [comment]
@aberleider Thanks for the patch: this is a good start.

Would you mind moving the bibliography information into `doc/doxygen/references.bib`? We did this in #8505.

**Comment by aberleider** [comment]
> @aberleider Thanks for the patch: this is a good start.
> 
> Would you mind moving the bibliography information into `doc/doxygen/references.bib`? We did this in #8505.

Sure thing, I just added the references to `doc/doxygen/references.bib`. I hope it went smoothly.

File: doc/news/changes/minor/20190807Sanchez-Vizuet [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Add references to the introduction of step 51
```
Comment (bangerth) [inline doc/news/changes/minor/20190807Sanchez-Vizuet]: Take a look at the other files in this direction -- they usually start with something like "Changed:". The empty line you have below should contain the text `<br>`.

File: examples/step-51/doc/intro.dox [hunk 2]
```diff
@@ -60,10 +53,11 @@ element-level solutions.  This is known as the
...
 </ol>
 
-The above procedure also has a linear algebra interpretation and is referred to
-as static condensation. Let us write the complete linear system associated to
-the HDG problem as a block system with the discrete DG (cell interior)
-variables $U$ as first block and the skeleton (face) variables $\Lambda$ as the
+<h4> Relation with Static Condensation </h4>
+The above procedure also has a linear algebra interpretation---referred to
+as <i>static condensation</i>---that was exploited to reduce the size of the global linear system by Guyan in the context of continuous Finite Elements @cite G65, and by Fraeijs de Veubeke for mixed methods @cite F65. In the latter case (mixed formulation), the system reduction was achieved through the use of discontinuous fluxes combined with the introduction of an additional auxiliary <i>hybrid</i> variable that approximates the trace of the unknown at the boundary of every element. This procedure became known as hybridization and---by analogy---is the reason why the local discontinuous Galerkin method introduced by Cockburn  and collaborators eventually became known as the <i>hybridizable discontinuous Galerkin</i> (HDG) method.
```
Comment (bangerth) [inline examples/step-51/doc/intro.dox]: here and below, can you break lines at roughly 80 characters?

File: examples/step-51/doc/intro.dox [hunk 3]
```diff
@@ -29,16 +29,9 @@ for systems of equations in 2 or 3 dimensions.
...
-(see the references in the recent HDG overview article by
- N.C. Nguyen and J. Peraire:
-   <i>Hybridizable discontinuous Galerkin methods for partial differential
-   equations in continuum mechanics</i>, Journal of Computational Physics,
-   2012, 231:18, 5955-5988.
-   <a href="http://dx.doi.org/10.1016/j.jcp.2012.02.033">[DOI]</a>).
-The HDG method achieves
-this goal by formulating the mathematical problem using Dirichlet-to-Neumann
-mappings.  The partial differential equations are first written as a first
+by Cockburn and co-workers (see the references in the recent HDG overview article by Nguyen and Peraire @cite NP12).
```
Comment (bangerth) [inline examples/step-51/doc/intro.dox]: We might get merge conflicts -- I think someone already moved this reference to the .bib file.  But go ahead and we'll deal with it as it happens.

Comment (bangerth) [reply examples/step-51/doc/intro.dox]: The change that also addresses this place is #8476.

**Comment by tjhei** [comment]
As somebody else has made changes to the same files, you will need to resolve the conflicts. Additionally, you did not set up your username and email correctly:
Author: MyAccount <myaccount@MyAccounts-MacBook-Pro.local>

Can you talk to one of the developers to get this fixed?

**Comment by aberleider** [comment]
> Please address the changes mentioned above. We also try hard to pick a descriptive title for each pull request. Yours reads "Local branch", which does not tell the reviewer what this is about. You can edit it by clicking "edit" at the top right.

Hi! I have re-configured my username and account, renamed the pull request and (hopefully) resolved the file conflicts.

**Comment by tjhei** [comment]
Thank you, much better!
Would you mind talking to one of us to rewrite the history so that we can remove your earlier commits (they still have your incorrect author info)?

**Comment by tjhei** [comment]
You also include a file called .DS_Store, which should not be part of this pull request. See one of the developers for help on that...

File: doc/news/changes/minor/20190807Sanchez-Vizuet [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+Include the references in the bibliography
+
+(Tonatiuh Sanchez-Vizuet August 7, 2019)
```
Comment (tjhei) [inline doc/news/changes/minor/20190807Sanchez-Vizuet]: As Wolfgang said, this format is not quite correct. Use ``<br>`` and format the date exactly how the other files in the same folder are formatted.

**Comment by masterleinad** [comment]
Would you mind to squash your commits?

File: examples/step-51/doc/intro.dox [hunk 5]
```diff
@@ -29,13 +29,12 @@ for systems of equations in 2 or 3 dimensions.
...
-this goal by formulating the mathematical problem using Dirichlet-to-Neumann
-mappings.  The partial differential equations are first written as a first
-order system, and each field is then discretized via a DG method.  At this
-point, the  single-valued "trace" values on the skeleton of the
+by Cockburn and co-workers (see the references in the recent HDG overview 
+article by Nguyen and Peraire @cite Ngu2012).
+
+The HDG method achieves this goal by formulating the mathematical problem using
+Dirichlet-to-Neumann mappings.  The partial differential equations are first
+ written as a first order system, and each field is then discretized via a DG method.  At this point, the  single-valued "trace" values on the skeleton of the
```
Comment (bangerth) [inline examples/step-51/doc/intro.dox]: Can you break this line at around 80 characters? I can't actually see what's beyond that point when looking at the pull request :-)

File: examples/step-51/doc/intro.dox [hunk 6]
```diff
@@ -55,11 +54,22 @@ element-level solutions.  This is known as the
...
-The above procedure also has a linear algebra interpretation and is referred to
-as static condensation. Let us write the complete linear system associated to
-the HDG problem as a block system with the discrete DG (cell interior)
-variables $U$ as first block and the skeleton (face) variables $\Lambda$ as the
-second block:
+<h4> Relation with Static Condensation </h4>
+The above procedure also has a linear algebra interpretation---referred to
+as <i>static condensation</i>---that was exploited to reduce the size of the
+ global linear system by Guyan in the context of continuous Finite Elements
+ @cite G65, and by Fraeijs de Veubeke for mixed methods @cite F65. In the latter
```
Comment (bangerth) [inline examples/step-51/doc/intro.dox]: Oh, these are *old*!!

File: examples/step-51/doc/intro.dox [hunk 7]
```diff
@@ -55,11 +54,22 @@ element-level solutions.  This is known as the
...
+ case (mixed formulation), the system reduction was achieved through the use of
+ discontinuous fluxes combined with the introduction of an additional auxiliary
+ <i>hybrid</i> variable that approximates the trace of the unknown at the
+ boundary of every element. This procedure became known as hybridization 
+and---by analogy---is the reason why the local discontinuous Galerkin method
```
Comment (bangerth) [inline examples/step-51/doc/intro.dox]: While you're here, editing this file again, can you remove the spaces that are at the beginning of some but not all of the lines above?

**Comment by aberleider** [comment]
> Would you mind to squash your commits?

Hi! I think I found the way to do it, but it was a little bumpy. I hope I did not break anything on the way.

File: examples/step-51/doc/intro.dox (hunk 6 above)
Comment (aberleider) [reply examples/step-51/doc/intro.dox]: > Oh, these are _old_!!
Just a little bit. The paper by Guyan is fascinating: it is hardly more than half a page long, but the idea is crystal clear. It seems almost unthinkable now to be able to introduce such a remarkable idea in so few lines.

File: examples/step-51/doc/intro.dox (hunk 5 above)
Comment (aberleider) [reply examples/step-51/doc/intro.dox]: Sure, I think I've cut all the super long lines. It should be better now.

File: examples/step-51/doc/intro.dox (hunk 7 above)
Comment (aberleider) [reply examples/step-51/doc/intro.dox]: With some luck I caught all the rouge spaces now.

**Comment by aberleider** [comment]
> @aberleider -- that's a really nice addition to the introduction of step-51! Thanks for finding these references and writing both eloquently and informed about the subject!
> 
> We've left a couple of comments. Find me (or one of the others) and we'll go through how to address them!

Thanks, I've tried my best to address the comments, but I'm still struggling to navigate the git waters. Hopefully I have not broken too many things in the attempt.

**Comment by drwells** [comment]
@aberleider The changes look good but I am concerned that this PR's history has some issues that need to be cleaned up. Would you be okay with me creating a new PR that consists of your commit on a different branch without the additional duplicated commits?

**Comment by aberleider** [comment]
@drwells Soinds good to me. Thanks for your help and sorry about the mess!

**Comment by drwells** [comment]
Superceded by #8549.

**Pull request closed without merging**